### PR TITLE
Address API 50 breaking changes to backup responses.

### PR DESCRIPTION
**Pull request by rjstires**
## Purpose
Address API release 50 breaking changes relating to backup responses.

https://github.com/linode/linode-api-docs/pull/46

## Notes
I added and used Ramda, a functional programming library for Javascript. If this is unreadable to either of you, give me the word and I'll replace it with a bunch of conditionals. I wanted to gauge reactions.

TL;DR ~ Create a list of backups, find in list by ID.

## Gold-platting
- Updated SummaryPage class and filename to BackupSummaryPage.

### CODE REVIEW

File: src/linodes/linode/backups/layouts/BackupPage.js [hunk 1]
```diff
@@ -1,13 +1,48 @@
 import PropTypes from 'prop-types';
 import React from 'react';
 import { connect } from 'react-redux';
-import { compose } from 'redux';
+// import { compose } from 'redux';
```
Comment (asauber) [inline src/linodes/linode/backups/layouts/BackupPage.js]: stray comment

File: src/linodes/linode/backups/layouts/BackupPage.js [hunk 2]
```diff
@@ -1,13 +1,48 @@
...
+import curry from 'ramda/src/curry';
+import propEq from 'ramda/src/propEq';
+
+const currentSnapshot = path(['snapshot', 'current']);
+
+const inProgressSnapshot = path(['snapshot', 'in_progress']);
+
+export const findById = curry((id, list) => list.find(propEq('id', id)));
+
+export const backupsToList = (backups) => compose(
```
Comment (displague) [inline src/linodes/linode/backups/layouts/BackupPage.js]: This should come with a line of shaping comments..  What is expected to go in, what comes out.

Comment (rjstires) [reply src/linodes/linode/backups/layouts/BackupPage.js]: I agree wholeheartedly, but this introduces some questions. JSDOC fails miserably at documenting functions which return functions. I can use Haskell style documentation, but I don't know how helpful that's going to be. Truly, this is a case for TypeScript or Flow.

With that said, I'll get some documentation in here for everyone.

**Comment by rjstires** [comment]
After additional consideration I've removed the use of Ramda from this PR pending a larger conversation with the team. There's really no use into shifting the style of the codebase until there's 100% buy-in from the team. Not saying anyone put up a fight, but it's not the right time.

**Pull request merged**